### PR TITLE
Build: remove `append_conf` _magic_ from MkDocs

### LINKED ISSUE #8529: mkdocs: Values with special `!` syntax are set to null on mkdocs.yml
## Details

* Read the Docs project URL: https://readthedocs.org/projects/openvidu/
* Build URL (if applicable): https://readthedocs.org/projects/openvidu/builds/14834239/
* Read the Docs username (if applicable):

Our project is using:

* MkDocs 1.2.2
* [Extra Context](https://www.mkdocs.org/dev-guide/themes/#extra-context) option in `mkdocs.yml`, with the [ENV syntax](https://www.mkdocs.org/user-guide/configuration/#environment-variables) to read environment variables, making the `READTHEDOCS` vars available to the template context:

    ```
    extra:
      READTHEDOCS: !ENV [READTHEDOCS, "(no var)"]
      READTHEDOCS_VERSION: !ENV [READTHEDOCS_VERSION, "(no var)"]
      READTHEDOCS_PROJECT: !ENV [READTHEDOCS_PROJECT, "(no var)"]
    ```

* Attempting to read the RtD environment variables as explained in [Build environment](https://docs.readthedocs.io/en/stable/builds.html#build-environment) and [How do I change behavior when building with Read the Docs?](https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-behavior-when-building-with-read-the-docs).

## Expected Result

1. The `cat mkdocs.yml` build step should show proper values for env vars `READTHEDOCS`, `READTHEDOCS_VERSION`, and `READTHEDOCS_PROJECT`.

2. A test html file with these contents:

    ```
    console.log("config.extra: {{ config.extra }}");
    console.log("config.extra.READTHEDOCS: {{ config.extra.READTHEDOCS }}");
    console.log("config.extra.READTHEDOCS_VERSION: {{ config.extra.READTHEDOCS_VERSION }}");
    ```

   should print something similar to this in the browser console:

    ```
    config.extra: {'READTHEDOCS': True, 'READTHEDOCS_VERSION': 1.2.3, 'READTHEDOCS_PROJECT': openvidu}
    config.extra.READTHEDOCS: True
    config.extra.READTHEDOCS_VERSION: 1.2.3
    ```

## Actual Result

1. The `cat mkdocs.yml` build step shows that `READTHEDOCS`, `READTHEDOCS_VERSION`, and `READTHEDOCS_PROJECT` are all `null`:

    ```
    extra:
      READTHEDOCS: null
      READTHEDOCS_PROJECT: null
      READTHEDOCS_VERSION: null
    ```

2. The browser console shows this output:

    ```
    config.extra: {'READTHEDOCS': None, 'READTHEDOCS_PROJECT': None, 'READTHEDOCS_VERSION': None}
    config.extra.READTHEDOCS: None
    config.extra.READTHEDOCS_VERSION: None
    ```

## Extra

Question / potential extra bug in the documentation: the technique shown in [How do I change behavior when building with Read the Docs?](https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-behavior-when-building-with-read-the-docs) would not really work with MkDocs, would it?

The sample is:

```
{% if READTHEDOCS %}
Woo
{% endif %}
```

but in MkDocs the `READTHEDOCS` is not directly available, is it expected to be? (sorry to ask here but is just a yes/no question that seems very relevant given the topic of the issue)

### PR DESCRIPTION
**Pull request by humitos**
Delete all the magic around MkDocs YAML that processes the `mkdocs.yml` file and define `readthedocs` theme automatically based on a feature flag, `MKDOCS_THEME_RTD`.

This PR removes:

- automatically defining `readthedocs` as theme when `MKDOCS_THEME_RTD` feature flag is defined.
- processing `mkdocs.yml` to add internal JS and CSS (embed and analytics) files automatically

This is another step forward on removing all the magic we perform on behalf the users and being more explicit about how to configure each doctool.

Reference:

* https://github.com/readthedocs/addons/issues/72#issuecomment-1926647293

Closes #8529

### CODE REVIEW

File: readthedocs/projects/signals.py
Comment (stsewd) [inline readthedocs/projects/signals.py:33]:
```suggestion
    We removed all the `mkdocs.yml` manipulation that set the theme to `readthedocs` if
```

File: readthedocs/projects/signals.py
Comment (stsewd) [inline readthedocs/projects/signals.py:48]: Shouldn't this logic always run? We are removing the old injection logic completely in this PR, if an old project is build, they won't have the old injections nor the new addons.

Comment (humitos) [reply readthedocs/projects/signals.py:48]: We have only a few projects on commercial using this feature only, and we plan to contact them telling how to migrate their configuration. Basically, they just need to define `theme.name: readthedocs` in their `mkdocs.yml` file.

I used this date here as an example, but we should give them more time than 20 days 😄  to migrate. We will contact them via email and write a blog post about this as well before merging this PR.

**Review by ericholscher — COMMENTED** [review]
This looks like a solid change, and a big step forward for Addons & our new build approach ❤️ Not 👍'ing yet until we make a full plan for rolling it out to unblock it.

File: readthedocs/projects/signals.py
Comment (humitos) [reply readthedocs/projects/signals.py:48]: I re-read your comment and I think you are right 👍🏼 . I updated this code to remove the datetime limitation.

**Comment by humitos** [comment]
**Merge this PR on April 15th** and deploy it after it.

**Comment by humitos** [comment]
This PR was reviewed but it doesn't have an approval yet. I re-checked the changes and it seems it's fine to be merged and deploy. Please, take a final look at it today since we need to merge and deploy it today/tomorrow as we published in our blog.

File: readthedocs/doc_builder/backends/mkdocs.py
Comment (stsewd) [inline readthedocs/doc_builder/backends/mkdocs.py:180]: I think we can remove all code related to `SafeLoader` and just call safe_load now.

**Review by stsewd — APPROVED** [review]

File: readthedocs/doc_builder/backends/mkdocs.py
Comment (humitos) [reply readthedocs/doc_builder/backends/mkdocs.py:180]: We can't yet because it's used to get the final doctype for the version at https://github.com/readthedocs/readthedocs.org/blob/8440aab6a0be1db5e8425d5bb96dee2802e164b8/readthedocs/doc_builder/backends/mkdocs.py#L66-L68

This code will be eventually removed since the doctype doesn't have too much sense anymore now that we support many doctools more. We should probably need to have a conversation about how to delete doctypes completely in a safe way.

Comment (stsewd) [reply readthedocs/doc_builder/backends/mkdocs.py:180]: yeah, but don't think you need the custom loader to just read that setting.

Comment (stsewd) [reply readthedocs/doc_builder/backends/mkdocs.py:180]: No if you call `safe_load()`

Comment (stsewd) [reply readthedocs/doc_builder/backends/mkdocs.py:180]: The custom loader was added to don't set those values to `None` after reading them (we were using `safe_load` previously).

Comment (humitos) [reply readthedocs/doc_builder/backends/mkdocs.py:180]: Cool. Good point! I will delete this chunk of code then 👍🏼

**Pull request merged**
